### PR TITLE
tests/lib/fakestore: add support for tracking channels

### PR DESCRIPTION
**Pull request by valentindavid**
Cherry-pick of eb2a9560e989335f9d5cb5b010b0b330886155e8 from https://github.com/canonical/snapd/pull/14305 in `fde-manager-features` branch.

### CODE REVIEW

File: tests/lib/fakestore/store/store.go [hunk 1]
```diff
@@ -447,6 +467,18 @@ func (s *Store) collectSnaps(bs asserts.Backstore) (map[string]*revisionSet, err
 
 		snaps[info.Name].add(snap.R(info.Revision), fn)
 
+		digest, _, err := asserts.SnapFileSHA3_384(fn)
```
Comment (andrewphelpsj) [inline tests/lib/fakestore/store/store.go:470]: The above `info` already carries the digest.

**Review by andrewphelpsj — COMMENTED** [review]
Thanks! I think adding a test here for the channel functionality would be good.

File: tests/lib/fakestore/store/store.go [hunk 2]
```diff
@@ -920,3 +955,40 @@ func findSnapRevision(snapDigest string, bs asserts.Backstore) (*asserts.SnapRev
...
+	dataPath := filepath.Join(cr.rootDir, snapDigest)
+	fd, err := os.Open(dataPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		} else {
+			return nil, err
+		}
+	} else {
+		sc := bufio.NewScanner(fd)
```
Comment (andrewphelpsj) [inline tests/lib/fakestore/store/store.go:987]: Missing a `defer fd.Close()` in here.

File: tests/lib/fakestore/store/store.go [hunk 3]
```diff
@@ -179,7 +197,9 @@ type essentialInfo struct {
 	Base        string
 }
 
-func snapEssentialInfo(fn, snapID string, bs asserts.Backstore) (*essentialInfo, error) {
+var errInfo = errors.New("cannot get info")
```
Comment (valentindavid) [inline tests/lib/fakestore/store/store.go:200]: Not sure what this is. I have to remove it.

File: tests/lib/fakestore/store/store_test.go [hunk 4]
```diff
@@ -423,6 +425,16 @@ func (s *storeTestSuite) makeAssertions(c *C, snapFn, name, snapID, develName, d
 	c.Assert(err, IsNil)
 }
 
+func (s *storeTestSuite) addToChannel(c *C, snapFn, channel string) {
+	dgst, _, err := asserts.SnapFileSHA3_384(snapFn)
+	c.Assert(err, IsNil)
+
+	f, err := os.OpenFile(filepath.Join(s.store.blobDir, "channels", dgst), os.O_CREATE|os.O_WRONLY, 0644)
```
Comment (andrewphelpsj) [inline tests/lib/fakestore/store/store_test.go:432]: Missing a `f.Close` here.

**Review by andrewphelpsj — APPROVED** [review]
Thank you!

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

**Comment by andrewphelpsj** [comment]
This is an updated to the fake store, and none of the failing tests use the fake store.
```
google-nested:ubuntu-22.04-64:tests/nested/manual/uc-update-assets-secure-add-sbat
google-nested:ubuntu-24.04-64:tests/nested/manual/hybrid-remodel
google-nested:ubuntu-20.04-64:tests/nested/manual/core20-new-snapd-does-not-break-old-initrd:startwithnew
google-nested:ubuntu-24.04-64:tests/nested/manual/remodel-min-size
```

**Pull request merged**